src/goto-checker/symex_bmc_incremental_one_loop.cpp

```diff
@@ -137,27 +137,7 @@ bool symex_bmc_incremental_one_loopt::resume(
 }
 void symex_bmc_incremental_one_loopt::log_unwinding(unsigned unwind)
 {
-  const std::string unwind_num = std::to_string(unwind);
-  switch(output_ui)
-  {
-  case ui_message_handlert::uit::PLAIN:
-  {
-    log.statistics() << "Current unwinding: " << unwind_num << messaget::eom;
-    break;
-  }
-  case ui_message_handlert::uit::XML_UI:
-  {
-    xmlt xml("current-unwinding");
-    xml.data = unwind_num;
-    log.statistics() << xml << messaget::eom;
-    break;
-  }
-  case ui_message_handlert::uit::JSON_UI:
-  {
-    json_objectt json;
-    json["currentUnwinding"] = json_numbert(unwind_num);
-    log.statistics() << json << messaget::eom;
-    break;
-  }
-  }
+  xmlt data{"current-unwinding"};
+  data.data = std::to_string(unwind);
+  log.statistics() << data;
```

Comment:
It could be a bit surprising that the result here is converted to a different format. Could we introduce a `structured_messaget` type as suggested by @smowton? It could well be implemented via `xmlt` for the time being (i.e., contain a member `xmlt` and just delegate to that). Then we could add a `print(const structured_messaget &msg)` method to `ui_message_handlert` which converts a `structured_messaget` to the appropriate format. The existing `print()` methods could be left unchanged.